sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatch.java

```diff
@@ -79,13 +79,11 @@ public static RowBasedKeyValueBatch allocate(StructType keySchema, StructType va
     boolean allFixedLength = true;
     // checking if there is any variable length fields
     // there is probably a more succinct impl of this
-    for (String name : keySchema.fieldNames()) {
-      allFixedLength = allFixedLength
-              && UnsafeRow.isFixedLength(keySchema.apply(name).dataType());
+    for (StructField field: keySchema.fields()) {
```

Comment:
Keep the space on both sides of the ":" in the for each loop. 
```suggestion
    for (StructField field : keySchema.fields()) {
```

Reply:
Fixed.